### PR TITLE
Keysend Messaging

### PR DESCRIPTION
**Pull request by deregs**
Implemented so far:
- send keysend messages through lnd
- send messages through c-lightning

To do:
- look into displaying received keysend messages in a new keysend page? Maybe leave this for a separate pr.

### CODE REVIEW

File: backends/CLightningREST.ts [hunk 1]
```diff
@@ -120,7 +120,8 @@ export default class CLightningREST extends LND {
     sendKeysend = (data: any) =>
         this.postRequest('/v1/pay/keysend', {
             pubkey: data.pubkey,
-            amount: Number(data.amt && data.amt * 1000)
+            amount: Number(data.amt && data.amt * 1000),
+            label: data.label
```
Comment (kaloudis) [inline backends/CLightningREST.ts:124]: remove the `label` field please. this is used for internal labeling, not keysend

**Comment by kaloudis** [comment]
Don't modify any of the locale files other than `en.json` please

File: backends/CLightningREST.ts (hunk 1 above)
Comment (deregs) [reply backends/CLightningREST.ts]: done

File: views/Send.tsx [hunk 2]
```diff
@@ -224,12 +226,14 @@ export default class Send extends React.Component<SendProps, SendState> {
...
         const { TransactionsStore, navigation } = this.props;
-        const { destination, maxParts, maxShardAmt, feeLimitSat } = this.state;
+        const { destination, maxParts, maxShardAmt, feeLimitSat, message } =
+            this.state;
 
         if (RESTUtils.supportsAMP()) {
             TransactionsStore.sendPayment({
                 amount: satAmount.toString(),
                 pubkey: destination,
+                message: message,
```
Comment (kaloudis) [inline views/Send.tsx:236]: if a property and the variable you are assigning to it have the same name you don't have to repeat it. See https://alligator.io/js/object-property-shorthand-es6/

we'll make sure we catch this with our linter by added this rule https://eslint.org/docs/rules/object-shorthand

File: views/Send.tsx [hunk 3]
```diff
@@ -238,7 +242,8 @@ export default class Send extends React.Component<SendProps, SendState> {
         } else {
             TransactionsStore.sendPayment({
                 amount: satAmount.toString(),
-                pubkey: destination
+                pubkey: destination,
+                message: message
```
Comment (kaloudis) [inline views/Send.tsx:246]: same here

File: utils/Base64Utils.ts [hunk 4]
```diff
@@ -84,6 +84,9 @@ class Base64Utils {
             (memo: any, i: number) => memo + ('0' + i.toString(16)).slice(-2),
             ''
         );
+
+    utf8ToHexString = (hexString: string) =>
+        Buffer.from(hexString, 'utf8').toString('hex');
```
Comment (kaloudis) [inline utils/Base64Utils.ts:89]: would be good to add some tests for this

Comment (deregs) [reply utils/Base64Utils.ts:89]: added a few

**Review by kaloudis — APPROVED** [review]
LFG!

**Pull request merged**